src/encoded/static/components/graph.js

```diff
@@ -365,78 +455,71 @@ export class Graph extends React.Component {
     }
 
     handleDlClick() {
-        // Collect CSS styles that apply to the graph and insert them into the given SVG element
-        function attachStyles(el) {
-            let stylesText = '';
-            const sheets = document.styleSheets;
-
-            // Search every style in the style sheet(s) for those applying to graphs.
-            // Note: Not using ES5 looping constructs because these aren’t real arrays.
-            if (sheets) {
-                for (let i = 0; i < sheets.length; i += 1) {
-                    const rules = sheets[i].cssRules;
-                    if (rules) {
-                        for (let j = 0; j < rules.length; j += 1) {
-                            const rule = rules[j];
-
-                            // If a style rule starts with 'g.' (svg group), we know it applies to the graph.
-                            // Note: In some browsers, indexOf is a bit faster; on others substring is a bit faster.
-                            // FF(31)'s substring is much faster than indexOf.
-                            if (typeof (rule.style) !== 'undefined' && rule.selectorText && rule.selectorText.substring(0, 2) === 'g.') {
-                                // If any elements use this style, add the style's CSS text to our style text accumulator.
-                                const elems = el.querySelectorAll(rule.selectorText);
-                                if (elems.length > 0) {
-                                    stylesText += `${rule.selectorText} { ${rule.style.cssText} }\n`;
-                                }
-                            }
-                        }
+        // Get dimensions of the displayed graph to determine whether it might cause a larger PNG
+        // than we can support.
+        const { savedSvg } = this.cv;
+        const viewBox = savedSvg.attr('viewBox');
+        const visibleWidth = savedSvg.attr('width');
+        const visibleHeight = savedSvg.attr('height');
+        if (visibleHeight > MAX_DOWNLOAD_GRAPH_HEIGHT) {
+            // Show an alert for the graph being too big to download.
+            this.setState({ isHeightWarningVisible: true });
+        } else {
+            // Draw a copy of the displayed graph but without decorations into a hidden div at the end
+            // of the <body> tag so it can get copied into a <canvas> for downloading. Prepare by
+            // inserting a blank temporary SVG into the hidden div.
+            const graphDownloadContainer = document.getElementById('graph-download-container');
```

Comment:
I assume you used document.getElementById rather than document.querySelector because of speed.